### PR TITLE
SDIT-1148 Throw 403 error if call using client credentials only fails due to incorrect agency access

### PR DESCRIPTION
**Pull request by karenmillermoj**

### CODE REVIEW

File: src/main/java/uk/gov/justice/hmpps/prison/service/AgencyService.java [hunk 1]
```diff
@@ -218,10 +220,12 @@ public void verifyAgencyAccess(final String agencyId) {
         if (agencyIds.isEmpty()) {
             if (authenticationFacade.isClientOnly()) {
                 logClientUnauthorisedAccess(agencyId);
+                throw new AccessDeniedException(format("Client not authorised to access agency with id %s.", agencyId));
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/hmpps/prison/service/AgencyService.java:223]: this message seems slightly odd - clients aren't restricted by `agencyId` anyway?

**Review by petergphillips — APPROVED** [review]
LGTM

File: src/main/java/uk/gov/justice/hmpps/prison/service/AgencyService.java (hunk 1 above)
Comment (karenmillermoj) [reply src/main/java/uk/gov/justice/hmpps/prison/service/AgencyService.java:223]: If they don't have the override role then the client isn't allowed to access the method if it has VerifyAgencyAccess

**Review by petergphillips — APPROVED** [review]
LGTM

**Pull request merged**